nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/main/java/org/apache/nifi/snmp/processors/AbstractSNMPProcessor.java

```diff
@@ -389,30 +391,31 @@ private AbstractTarget createSnmpTarget(ProcessContext context) {
             break;
         }
 
-        if(version == SnmpConstants.version3) {
-            final String username = context.getProperty(SNMP_SECURITY_NAME).getValue();
-            final String authPassword = context.getProperty(SNMP_AUTH_PASSWORD).getValue();
-            final String privPassword = context.getProperty(SNMP_PRIV_PASSWORD).getValue();
-            final String authProtocol = context.getProperty(SNMP_AUTH_PROTOCOL).getValue();
-            final String privProtocol = context.getProperty(SNMP_PRIV_PROTOCOL).getValue();
+        AbstractTarget result;
+        if (version == SnmpConstants.version3) {
+            String username = context.getProperty(SNMP_SECURITY_NAME).getValue();
```

Comment:
That is mostly a personal preference as I find they are mostly not needed. I know that there's coding style guidelines for this in some Hadoop related projects and most of them don't use `final` for local variables or method parameters. So I adopted this style here. But if NiFi follows another style I'm happy to keep them.